.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,82 @@
+name: "CodeQL"
+
+on:
+  # Run the analysis once every 24 hours. The actual time does not matter
+  # that much, start with a time that allows for easier troubleshooting.
+  schedule:
+    - cron: '00 22 * * *'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp' ]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Checkout vcpkg
+        uses: actions/checkout@v2
+        with:
+          path: "build/vcpkg"
+          repository: "microsoft/vcpkg"
+          ref: "7e7bcf89171b7ed84ece845b1fa596a018e462f2"
```

Comment:
We might want to do something like [this](https://github.community/t/read-from-file-and-assign-to-env-variable/174936/2) to read the vcpkg commit from `ci/etc/vckpg-commit.txt`

(if we keep this action, that is)

Reply:
That cannot work, each step runs in its own docker container, you cannot pass variables from one to the next. But I can add a step that performs the checkout at the right SHA. PTAL